app/models/user.rb

```diff
@@ -0,0 +1,35 @@
+class User < ApplicationRecord
+  include Authenticatable
+  include Steppable
+
+  enum aasm_state: {
+    stage_1:  1,
+    stage_2:  10,
+    stage_3:  100,
+    stage_4:  1_000,
+    stage_5:  10_000
+  }
+
+  aasm do
+    state :stage_1, initial: true
+    state :stage_2
+    state :stage_3
+    state :stage_4
+    state :stage_5
+
+    event :complete_stage, after: :set_highest_aasm_state do
```

Comment:
Calling `current_user.complete_stage` will transition from the current stage to the one corresponding in the `transitions` call. As a callback afterward, potentially set the `User#highest_aasm_state` to remember the furthest page the User has been to (and ultimately redirect to it after submitting in the web app).